spec/features/hq/create_sponsor_spec.rb

```diff
@@ -0,0 +1,45 @@
+require 'rails_helper'
+
+RSpec.feature 'User enters new sponsor data', :type => :feature do
+
+  background do
+    i_sign_in_as_admin
+    an_agent_exists_with_username
+  end
+
+  scenario 'User clicks the New Sponsor button' do
+    when_i_register_new_sponsor
+
+    then_the_sponsor_should_be_saved
+    and_i_should_be_on :new_sponsor_page
+  end
+
+  scenario 'With invalid input' do
+    visit new_hq_sponsor_path
+    click_button 'Create and Add Another'
+    expect(page).to have_content 'Please fix the errors below and resubmit the form.'
+  end
+
+  def when_i_register_new_sponsor
+    FactoryGirl.create(:sponsor, city: 'Sigurdland')
```

Comment:
I guess this record is created only to have a city in the DB?

I think we should skip `FactoryGirl.create` here and use `New city name` text field.(this will create a new city)
Fill in all fields.
Then click `Create and Add Another` and fill in the form with a second sponsor and use the same city by selecting the selectors option.
Then click "Create Sponsor" and check that it redirects and shows the correct sponsor and details (all details).
Check that you are on sponsor_show and see "Sponsor successfuly created"
Check all field are correct.
Then go to index and check that both sponsors are present.

1st sponsor should use Organization, 2nd sponsor should use Branch
Also we should fill in all fields, not only the required.